src/libsyntax/diagnostics/plugin.rs

```diff
@@ -111,7 +111,7 @@ pub fn expand_register_diagnostic<'cx>(ecx: &'cx mut ExtCtxt,
     // overflow the maximum line width.
     description.map(|raw_msg| {
         let msg = raw_msg.as_str();
-        if !msg.starts_with("\n") || !msg.ends_with("\n") {
+        if !msg.starts_with('\n') || !msg.ends_with('\n') {
```

Comment:
@petrochenkov Ah I see. Indeed `msg.starts_with('\n')` [generates more code](https://godbolt.org/g/fn0THW). However, this should be something that rustc and/or LLVM that can be optimized out? The input is a constant ASCII character which all UTF-8-related branches should be elided.